src/v2/controllers/FormController.js

```diff
@@ -240,22 +240,36 @@ export default Controller.extend({
   },
 
   showFormErrors(model, error) {
+    let errorObj, idxStateError;
     model.trigger('clearFormError');
+    
     if (!error) {
       error = 'FormController - unknown error found';
       this.options.settings.callGlobalError(error);
     }
-    let errorObj;
+
+    if(error?.rawIdxState) {
+      error = error.rawIdxState;
+    }
+
     if (IonResponseHelper.isIonErrorResponse(error)) {
       errorObj = IonResponseHelper.convertFormErrors(error);
     } else if (error.errorSummary) {
       errorObj = { responseJSON: error };
     }
+    
+    // show error before updating app state.
     model.trigger('error', model, errorObj || { responseJSON: { errorSummary: String(error) } }, true);
+    idxStateError = Object.assign({}, error, {hasFormError: true});
+    // TODO OKTA-408410: Widget should update the state on every new response. It should NOT do selective update.
+    // For eg 429 rate-limit errors, we have to skip updating idx state, because error response is not an idx response.
+    if (Array.isArray(error?.neededToProceed) && error?.neededToProceed.length) {
+      this.handleIdxResponse(idxStateError);
+    }
```

Comment:
I thought this is only applicable when `IonResponseHelper.isIonErrorResponse(error)` is `true` ?

Reply:
Ideally widget should update the app state on every response.. also after idx.js error responses will always be in idx format.